language-plutus-core/src/Language/PlutusCore/Normalize/Internal.hs

```diff
@@ -203,18 +193,6 @@ substNormalizeTypeM ty name = withExtendedTypeVarEnv name ty . normalizeTypeM
 normalizeTypesInM
     :: (HasUnique (tyname ann) TypeUnique, MonadQuote m)
     => Term tyname name ann -> NormalizeTypeT m tyname ann (Term tyname name ann)
-normalizeTypesInM = go where
+normalizeTypesInM = transformMOf termSubterms (traverseOf termSubtypes normalizeReturnType) where
```

Comment:
I suggest to inline the definition of `normalizeReturnType` here (and probably remove `traverseOf`).

Reply:
Sure - I kind of like `traverseOf` because otherwise I have to break the abstraction barrier and remember that a traversal is a function :grimacing:

Reply:
Fair enough.